src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -838,7 +842,7 @@ public SolutionState WithHasAllInformation(ProjectId projectId, bool hasAllInfor
 
             Debug.Assert(this.ContainsProject(projectId));
 
-            var oldProject = this.GetProjectState(projectId);
+            var oldProject = this.GetProjectState(projectId)!;
```

Comment:
This is a really sneaky way to make the var a non-null reference type. Is there some more explicit way we can get project state with an assertion of nullability? Maybe add an internal/private api like `GetProjectStateNotNull(projectId)` or some more clever named thing.

Reply:
GetProjectState is already internal so we could just change it. @ryzngard thoughts if it's better to just change it in this PR versus do that in another one? Was generally hoping to keep the behavior changes to a minimum in this.

Reply:
Actually: having this annotated how it "works now" and then using that to do a separate pass to change behavior actually is the right way to do it since you can review each transformation separately.

Reply:
I agree, another review would be better